lib/services/presets.go

```diff
@@ -502,27 +502,59 @@ func bootstrapRoleMetadataLabels() map[string]map[string]string {
 func defaultAllowRules() map[string][]types.Rule {
 	return map[string][]types.Rule{
 		teleport.PresetAuditorRoleName: {
+			types.NewRule(types.KindSession, RO()),
+			types.NewRule(types.KindEvent, RO()),
 			types.NewRule(types.KindSessionTracker, RO()),
+			types.NewRule(types.KindClusterAlert, RO()),
 			types.NewRule(types.KindInstance, RO()),
 		},
 		teleport.PresetEditorRoleName: {
```

Comment:
If we add everything, we might as well just use the default rules for the Role and iterate over it.
And document that behaviour.

What do you think?

Right now, we have two lists and a test ensuring they are the same

Reply:
That's fine with me. Is that reasonable, or is there a reason that this is separate at present?

Reply:
I think we should be good.
The initial goal was only for newly added resources/rules that we wanted to backport on existing clusters.
If we add everything, then we might as well just use the current set of rules for backporting.

However, there might be some cases where this is not intended.
We can see multiple new resources, created after the `defaultAllowRules` definition, that were not added.
People either forget or maybe didn't want to add them.

Eg, https://github.com/gravitational/teleport/pull/18397
> I'll be conservative here and not retrofit permissions into editor, a fact to be called out in our to-be-written user-facing documentation.

@codingllama What was the reason for not backporting the new permission that you added?
Would it be fine to add them now?

Reply:
Sorry, I can't find the comment above, what are we talking about?

Backport what to which branch? From the date of the PR, I would expect those verbs to be everywhere they need to be by now.

Reply:
> Backport what to which branch?

Sorry, ignore "backport". I meant "backfill" (as in add new permissions to existing roles definitions in older clusters)

Reply:
OK, I think I follow this better now. Idk why we have separate lists, looks off to me to too.

In the specific case of Device Trust, and I suppose many others, I found the role definition first and didn't know this second list existed.

As for changing presets, we do that often and they do get automatically updated (IIRC), so I suppose it's not a problem?

> It's part of the PR description.

🤦

Reply:
> As for changing presets, we do that often and they do get automatically updated

I don't think they do, unless you add them to `defaultAllowRules`
That is the whole purpose of that list.

There are some new Roles that are upserted everytime, but have specific labels, and `editor,access,auditor` is not in that scope.

So, trying to reach a conclusion here: let's re-use the initial set of rules to backfill the preset roles.

Reply:
It seems reasonable to me, but I'm clearly not the person to ask.  ;)

Reply:
@jakule @capnspacehook Do you see an issue with backfilling the preset roles with newly added rules?
So, applying the same logic we have today for `defaultAllowRules` but iterating over the list of the role's Allow Rules, instead of using a subset of those rules (as we do now).

Reply:
Makes sense to me

Reply:
Okay, made this change!